### ISSUE TITLE
Unable to install nuget package with custom seq app

### ISSUE THREAD

**Issue by per-samuelsson** [datalust/seq-tickets#1974]
**Describe the bug**
I've built a simple Seq app and uploaded the NuGet package to GitHubs package feed.
I've configured Seq to consume packages from above feed, but when I try to `INSTALL FROM NUGET` in the web-interface, I get an error, and when checking the log, I see this:

```
[08:09:23 INF] NuGet v3 "GET" "https://nuget.pkg.github.com/octocat/index.json"
[08:09:23 INF] NuGet v3 "GET" "https://nuget.pkg.github.com/octocat/foo/index.json"
[08:09:24 INF] NuGet v3 "GET" "https://nuget.pkg.github.com/octocat/foo/1.6.0.json"
[08:09:24 INF] NuGet v3 "GET" "https://nuget.pkg.github.com/octocat/download/foo/1.6.0/foo.1.6.0.nupkg"
[08:09:25 ERR] Unable to install package "foo" v"1.6" from "nugetfeed-41"
System.IO.InvalidDataException: End of Central Directory record could not be found.
   at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory()
   at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leaveOpen, Encoding entryNameEncoding)
   at System.IO.Packaging.ZipPackage..ctor(String path, FileMode packageFileMode, FileAccess packageFileAccess, FileShare share)
   at System.IO.Packaging.Package.Open(String path, FileMode packageMode, FileAccess packageAccess, FileShare packageShare)
   at Seq.Server.Features.Installation.LightweightPackageInstaller.Install(String packageFile, String directory)
   at Seq.Server.Features.Installation.AppPackageStore.GetOrInstallAsync(NuGetFeed feed, String packageId, String version, Boolean force)
   at Seq.Server.Features.Installation.AppPackageInstaller.GetOrInstallAsync(String nuGetFeedId, String packageId, String version, Boolean force)
   at Seq.Server.Features.Installation.AppPackageInstaller.InstallPackageAsync(String nuGetFeedId, String packageId, String version)
   at Seq.Server.Web.Api.AppsController.InstallPackage(String feedId, String packageId, String version)
```

**Environment (please complete the following information):**
 - OS: Under WSL2 on Windows, I'm running a Seq docker container.
 - Seq Version: `Seq 2023.1.9133`
 - Using Docker? Yes

**Additional context**
* Creating a simple .NET 7.0 Console app (under WSL2, `Ubuntu-20.04`), I can consume the exact same package without a problem.
* I can see that the Seq server actually download the package (I see the download count on GH increase by 1).
* I've replaced the real GH organisation name with `octocat` and the package name with `foo`.
* This is on a local dev-env, I haven't checked if the same would occur on our paid Seq instance.

Any help with be highly appreciated!

**Comment by liammclennan** [comment]
Hi @per-samuelsson 

That error message indicates that the NuGet package is not a valid zip file. If you find it on the filesystem where Seq has downloaded it (in the `Packages` directory) you should be able to confirm that the zip archive is not valid. 

The simplest solution is to add a directory as a NuGet feed, manually place the NuGet package in that directory and install from there.

**Comment by per-samuelsson** [comment]
Hi @liammclennan,

Thanks for the fast and accurate answer. It certainly pointed me in a direction closer to the actual problem, although I still don't understand exactly why my approach fail.

If I enter the container and browse the `Packages` folder, it actually shows there there isn't even any file there. But other packages, from the public feed, are working fine.

```
/Packages/nugetfeed-1/Seq.App.Timeout/2.0.0:
  Seq.App.Timeout.2.0.0.nupkg
/Packages/nugetfeed-41/Foo
  (empty)
```

That certainly explain why the End of Central Directory record can't be found at least. 😄

**Comment by liammclennan** [comment]
Hi @per-samuelsson 

It could be that downloading from the github feed fails because the server returns a redirect. Are you able to check (with curl or similar) if github returns a redirect for requests for packages?

**Comment by per-samuelsson** [comment]
Hi @liammclennan 

I took me some time to find out how, considering the private GH feed don't reveal exactly how to fetch assets using a client other than `nuget`, but by studying Seq-logs, I found the resolved package URL and was able to test it using `curl`.

And, yes, confirmed, it does in fact redirect. So then that's the problem (I remember seeing issues that you are working with redirects).

While at it, I can also confirm that using your suggested approach to host the nuget package in a local feed on the server, and configure Seq to consume that, I can use the app (although it's somewhat inconvenient of course 😸).

With that, I think it's fine you close this one as a dup of redirection-issues.
Big thanks for the support 👍
